providers/salesforce/register.go

```diff
@@ -0,0 +1,231 @@
+		Result:          result,
+		Status:          common.RegistrationStatusSuccess,
+	}, err
+}
+
+func (c *Connector) register(
+	ctx context.Context,
+	params *RegistrationParams,
+) (*ResultData, error) {
+	result := &ResultData{}
+
+	eventChannel, err := c.createEventChannel(ctx, params)
+	if err != nil {
+		return result, fmt.Errorf("failed to create event channel: %w", err)
+	}
+
+	result.EventChannel = eventChannel
+
+	namedCred, err := c.createNamedCredential(ctx, params)
+	if err != nil {
+		return result, fmt.Errorf("failed to create named credential: %w", err)
+	}
+
+	result.NamedCredential = namedCred
+
+	evtCfg, err := c.createEventRelayConfing(
+		ctx,
+		params,
+		namedCred.DestinationResourceName(),
+		eventChannel.FullName,
+	)
+	if err != nil {
+		return result, fmt.Errorf("failed to create event relay config: %w", err)
+	}
+
+	result.EventRelayConfig = evtCfg
+
+	if err := c.RunEventRelay(ctx, evtCfg); err != nil {
+		return result, fmt.Errorf("failed to run event relay: %w", err)
+	}
+
+	return result, nil
+}
+
+func (c *Connector) createEventChannel(ctx context.Context, params *RegistrationParams) (*EventChannel, error) {
+	channelName := GetChannelName(params.UniqueRef)
+
+	channel := &EventChannel{
+		FullName: channelName,
+		Metadata: &EventChannelMetadata{
+			ChannelType: "data",
+			Label:       params.UniqueRef,
+		},
+	}
+
+	return c.CreateEventChannel(ctx, channel)
+}
+
+func (c *Connector) createNamedCredential(ctx context.Context, params *RegistrationParams) (*NamedCredential, error) {
+	namedCred := &NamedCredential{
+		FullName: params.UniqueRef,
+		Metadata: &NamedCredentialMetadata{
+			GenerateAuthorizationHeader: true,
+			Label:                       params.Label,
+
+			// below are legacy fields
+			Endpoint:      params.AwsArn,
+			PrincipalType: "NamedUser",
+			Protocol:      "NoAuthentication",
+		},
+	}
+
+	return c.CreateNamedCredential(ctx, namedCred)
+}
+
+func (c *Connector) createEventRelayConfing(
+	ctx context.Context,
+	params *RegistrationParams,
+	destinationResource string,
+	channelName string,
+) (*EventRelayConfig, error) {
+	config := &EventRelayConfig{
+		FullName: params.UniqueRef,
+		Metadata: &EventRelayConfigMetadata{
+			DestinationResourceName: destinationResource,
+			EventChannel:            channelName,
+		},
+	}
+
+	return c.CreateEventRelayConfig(ctx, config)
+}
+
+func IsCustomObject(objName string) bool {
+	return strings.HasSuffix(objName, "__c")
+}
+
+func GetRawObjectName(objName string) string {
+	return RemoveSuffix(objName, 3) //nolint:mnd
+}
+
+func GetChangeDataCaptureEventName(objName string) string {
+	if IsCustomObject(objName) {
+		return GetRawObjectName(objName) + "__ChangeEvent"
+	}
+
+	return objName + "%ChangeEvent"
+}
+
+func GetChannelName(rawChannelName string) string {
+	return rawChannelName + "__chn"
+}
+
+func RemoveSuffix(objName string, suffixLength int) string {
+	return objName[:len(objName)-suffixLength]
```

Comment:
Seems like this value might accidentally go negative which might panic. Might want to bounds check the values.